ddl/table_split_test.go

```diff
@@ -51,5 +53,11 @@ func (s *testDDLTableSplitSuite) TestTableSplit(c *C) {
 	cache := store.(kvStore).GetRegionCache()
 	loc, err := cache.LocateKey(tikv.NewBackoffer(context.Background(), 5000), regionStartKey)
 	c.Assert(err, IsNil)
-	c.Assert(loc.StartKey, BytesEquals, []byte(regionStartKey))
+	if bytes.Compare(loc.StartKey, []byte(regionStartKey)) != 0 {
+		// Region cache may be out of date, so we need to drop this expired region and load it again.
+		cache.DropRegion(loc.Region)
+		loc, err := cache.LocateKey(tikv.NewBackoffer(context.Background(), 5000), regionStartKey)
```

Comment:
What happens when the result is stale again, should it be a while condition rather than if condition?

Reply:
Why is it stale again?

Reply:
> The LocateKey gets the region from region cache.
> But region cache may be out of date.